### ISSUE TITLE
Duplicate file checking  ( request )

### ISSUE THREAD

**Issue by JohnDarkhorse** [pxgamer/vidox#14]
Would there be a possibility of getting some sort of low-resource duplicate file hash checking?

For instance, Joe uploads a video on Monday.

On Tuesday, Joe's coworker uploads the same video.

As it stands, there'd be two identical videos with two different IDs on the server.

With a hash checking system, anyone who uploaded a duplicate video could be shown the URL for the existing video.

**Comment by owenvoke** [comment]
Yes. It would be possible, can use something such as sha1_file() or hash using sha256.

I haven't really done much with this recently.

I'll have a look tomorrow, but if you want to add in hash checking, feel free to make a PR.

**Comment by JohnDarkhorse** [comment]
> but if you want to add in hash checking

I'm not very good at programming and would not even know where to start in adding a file hashing function, but I do know that de-duplication would be a welcome feature where video files are concerned
|
|
|
One would think that - keeping with the "single script" ( light weight ) concept - since we're not securing a bank transfer or ordering from an online retailer -, an outdated cryptographic hashing function could be used if it was more resource-efficient than current offerings.
|
|
|
Perhaps it could write to a text file like so . . .

`hashhashhashhashhashhash  #filename`

. . . where the filename is there for the webmaster, in case s/he decides to delete the video file, they can clear the hash from the list.

**Comment by owenvoke** [comment]
May be better to add it as a database column instead. Then it can just check if it's unique, if not, then it gives you the link to the old video.
